### PR TITLE
[Foundation] Correct NSRange equality typo and add unit tests for newly added NSRange behaviors

### PR DESCRIPTION
**Pull request by phausler**
This fixes <rdar://problem/32649690> as well as a few other minor cleanup issues discovered when testing. And most importantly adds unit tests for the new functionality of NSRange

### CODE REVIEW

File: test/stdlib/TestNSRange.swift [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    func testEquality() {
+        let r1 = NSRange(location: 1, length: 10)
+        let r2 = NSRange(location: 1, length: 11)
+        let r3 = NSRange(location: 2, length: 10)
+        let r4 = NSRange(location: 1, length: 10)
+
+        expectNotEqual(r1, r2)
+        expectNotEqual(r1, r3)
+        expectEqual(r1, r4)
+    }
```
Comment (parkera) [inline test/stdlib/TestNSRange.swift]: Maybe a equality test with NSNotFound too?

**Comment by phausler** [comment]
@swift-ci please test

**Comment by phausler** [comment]
hmm the failure does not seem to be part of NSRange:

```
/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/decl/protocol/conforms/fixit_stub_editor.swift:4:4: error: unexpected error produced: protocol members can only be marked unavailable in an @objc protocol
13:59:03   @available(iOS, unavailable)
13:59:03    ^
```

@swift-ci please test

**Comment by phausler** [comment]
@swift-ci please test

**Comment by phausler** [comment]
@swift-ci please test

**Pull request merged**